Overmap 4.6: Anomalous Activity

Erikafox opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

https://github.com/shiptest-ss13/Shiptest/assets/94164348/591dd5bc-c90a-40d0-a578-54b9a0abe90d

Anomalies will now spawn on planets and remain static there, posing a passive threat to anyone who comes across them. If you're feeling daring, you can use an analyzer and signaller, or an anomaly neutralizer, to harvest their cores, and put them in an R&D console for research points.

Anomaly effects have also been expanded, you'll have to find out how in game though.

As a side effect to make bluespace anomalies work, removes the noteleport flag on planets
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
More exploration content
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl: Bjarl, Ghilker, Comyx, Assorted /tg/ contributors
add: anomalies will now randomly spawn on planets
add: bioscrambler anomalies, they randomly change your organs and bodyparts
add: hallucination anomalies, they cause hallucinations
add: big anomalies, they're like normal anomalies but big.
fix: disk renaming should only work with pens now
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Erikafox committed:
Does The Thing

Erikafox commented:
further proof as to why I should not be allowed to code.

Erikafox committed:
linterinos?

Erikafox commented:
Things I forgot to do and need to do tomorrow:
Experiment with size more
make bioscramblers give a few status effects so you have an indication of what's happening
unfucker bioscrambler clone damage

what I might do:
Subtype all the anomalies to be /planetary and restore original wandering behavior to the main types
make anomalies move in an area around their spawn tile (I would need help figuring this out)

Erikafox committed:
update

Erikafox committed:
Update RockGenerator.dm

rye-rice reviewed (approved):
sprites are fine, theyre from tg

maybe make the bioscrambler either rarer or not do the effects immediatly, they could be hidden behind something and someone could get messed up with long term effects without realizing there was a bioscrambler nearby? maybe take this with a grain of salt since im just a spritetainer

Erikafox commented:
I can donk it down so that you just get the status effects at a long range. That's probably better for balance anyways

Erikafox committed:
Update anomalies_bioscrambler.dm

thgvr commented:
dunno about the bioscrambler anomaly

Erikafox committed:
Update anomalies.dm

Erikafox commented:
I'm gonna cut it and whip up a few more anomalies to diversify the pool

goober3 commented:
![image](https://github.com/shiptest-ss13/Shiptest/assets/118859017/5ac5d6aa-f774-4041-9e6f-e047a93b11d6)
hi erika i think you should maybe remove bluespace anomalies

Erikafox commented:
I would specify why that isn't a good reason but we're still in round

basically: skill issue

Erikafox committed:
workworkwork

Erikafox committed:
bhhhhhh

Erikafox committed:
h

Erikafox committed:
d

Erikafox committed:
as

Erikafox committed:
Update anomalies_static.dm

Erikafox committed:
Update anomalies_static.dm

Erikafox committed:
Update anomalies_static.dm

Erikafox committed:
Finishes the PR for real this time

bugfixes not included

Erikafox committed:
Merge branch 'master' into planetary-anomalies

Erikafox committed:
I could have sworn I removed this.

Erikafox committed:
Merge remote-tracking branch 'upstream/master' into planetary-anomalies

Erikafox committed:
Merge branch 'master' into planetary-anomalies

Erikafox committed:
Update WasteGenerator.dm

Erikafox committed:
Update WasteGenerator.dm

Erikafox committed:
Merge remote-tracking branch 'upstream/master' into planetary-anomalies

Erikafox committed:
Update SnowGenerator.dm

Erikafox committed:
unthrongles for the fourth time

MitztheKat commented on code/game/objects/effects/anomalies/anomalies_static.dm:
```diff
@@ -0,0 +1,68 @@
+	for(var/mob/living/carbon/looking in range(effectrange, src))
+		if (!HAS_TRAIT(looking, TRAIT_MINDSHIELD) && looking.stat != DEAD)
+			looking.adjustOrganLoss(ORGAN_SLOT_BRAIN, 10, 200)
+			playsound(looking, 'sound/effects/walkietalkie.ogg')
```
you need to give bigger warning for these.

MitztheKat commented on code/game/objects/effects/anomalies/anomalies_static.dm:
```diff
@@ -0,0 +1,68 @@
+	density = TRUE
+	aSignal = /obj/item/assembly/signaler/anomaly/tvstatic
+	effectrange = 6
+	pulse_delay = 6
```
too long-range, too fast.

Erikafox committed:
balance changes from tm

Erikafox committed:
more improvements

Erikafox committed:
Update anomalies_plasmasoul.dm

Erikafox committed:
Update anomalies_melter.dm

Erikafox committed:
update

Erikafox committed:
Merge remote-tracking branch 'upstream/master' into planetary-anomalies

Erikafox committed:
Merge remote-tracking branch 'upstream/master' into planetary-anomalies

Erikafox committed:
Update curse.dm

spockye commented:
please
please
please

decrease the spawn rate, and KILL the THONGLER

meemofcourse commented:
the throngler slayer

Erikafox committed:
update

BarteG44 commented:
walked like a 100 tiles straight north on a rock planet

counted 8 anomalies on the way

Erikafox commented:
with the current commit or not

I toned spawning *heavily* down in the last push.

BarteG44 commented:
> with the current commit or not
> 
> I toned spawning _heavily_ down in the last push.

from before

Erikafox committed:
reduces spawn chances More

Erikafox committed:
Update SandGenerator.dm

Erikafox committed:
Update lootdrop.dm

Erikafox committed:
Update lootdrop.dm

Erikafox committed:
more updates

pulse timing is now a range. Research goggles can tell you how much science is in an anomaly

Erikafox committed:
trying more changes

Erikafox committed:
cleans the code up more

Erikafox committed:
final adjustment

MarkSuckerberg commented on code/modules/mob/living/carbon/status_procs.dm:
```diff
@@ -94,3 +94,8 @@
+/mob/living/carbon/proc/break_random_bone() //this might work
+	var/obj/item/bodypart/B = get_bodypart()
+	for(B in bodyparts)
+		B.break_bone()
```
```suggestion
	var/obj/item/bodypart/limb = pick(bodyparts)
	limb.break_bone
```

MarkSuckerberg commented on code/modules/research/rdconsole.dm:
```diff
@@ -136,7 +136,7 @@ Nothing else in the console has ID requirements.
 			else
 				playsound(src, 'sound/machines/ping.ogg', 50, 3, -1)
-				visible_message("<span class='notice'>You insert [E] into a slot on the [src], producting [E.research] points from the extract's chemical makeup!</span>")
+				visible_message("<span class='notice'>[user] inserts [E] into a slot on the [src], producting [E.research] points from the extract's chemical makeup!</span>")
```
```suggestion
				visible_message("<span class='notice'>[user] inserts [E] into a slot on the [src]!</span>", "<span class='notice'>You insert [E] into a slot on the [src], producting [E.research] points from the extract's chemical makeup!</span>")
```

MarkSuckerberg commented on code/modules/research/rdconsole.dm:
```diff
@@ -155,7 +155,7 @@ Nothing else in the console has ID requirements.
 			else
 				playsound(src, 'sound/machines/ping.ogg', 50, 3, -1)
-				visible_message("<span class='notice'>You insert [E] into a slot on the [src], producting [E.research] points from the plant's genetic makeup!</span>")
+				visible_message("<span class='notice'>[user] inserts [E] into a slot on the [src], producting [E.research] points from the plant's genetic makeup!</span>")
```
```suggestion
				visible_message("<span class='notice'>[user] inserts [E] into a slot on the [src]!</span>", "<span class='notice'>You insert [E] into a slot on the [src], producting [E.research] points from the plant's genetic makeup!</span>")
```

MarkSuckerberg commented on code/modules/research/rdconsole.dm:
```diff
@@ -164,6 +164,18 @@ Nothing else in the console has ID requirements.
+			return
+
+		playsound(src, 'sound/machines/ping.ogg', 50, 3, -1)
+		visible_message("<span class='notice'>[user] inserts [anomaly] into a slot on the [src], producting [anomaly.research] points!</span>")
```
```suggestion
		visible_message("<span class='notice'>[user] inserts [anomaly] into a slot on the [src]!</span>", "<span class='notice'>You insert [anomaly] into a slot on the [src], producting [anomaly.research] points!</span>")
```

MarkSuckerberg commented on code/game/objects/effects/anomalies/anomalies_vortex.dm:
```diff
@@ -0,0 +1,69 @@
+			SSexplosions.medobj += O
+
+/obj/effect/anomaly/vortex/proc/grav(r, ex_act_force, pull_chance, turf_removal_chance)
+	for(var/t = -r, t < r, t++)
```
```suggestion
	for(var/t in -r to r)
```
a more descriptive var name would also be nice, I'm not fully sure what this means exactly

MarkSuckerberg commented on code/game/objects/effects/anomalies/anomalies_hallucination.dm:
```diff
@@ -0,0 +1,52 @@
+/obj/effect/anomaly/hallucination/proc/hallucination_pulse(turf/open/location, effectrange)
+	for(var/mob/living/carbon/human/near in view(location, effectrange))
+		// If they are immune to hallucinations.
+		if (HAS_TRAIT(near, SEE_TURFS) || (near.mind && HAS_TRAIT(near.mind, SEE_TURFS)))
```
is this the right trait?

Erikafox replied:
oop

MarkSuckerberg commented on code/game/objects/effects/anomalies/anomalies_bluespace.dm:
```diff
@@ -0,0 +1,119 @@
+	blueeffect.plane = FULLSCREEN_PLANE
+	blueeffect.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	M.client.screen += blueeffect
+	sleep(20)
```
```suggestion
	stoplag(20)
```

Erikafox committed:
Apply suggestions from code review

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: Bjarl <94164348+Bjarl@users.noreply.github.com>

Erikafox committed:
Update code/modules/mob/living/carbon/status_procs.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: Bjarl <94164348+Bjarl@users.noreply.github.com>

Erikafox committed:
Update code/game/objects/effects/anomalies/anomalies_bluespace.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: Bjarl <94164348+Bjarl@users.noreply.github.com>

Erikafox committed:
Update code/game/objects/effects/anomalies/anomalies_vortex.dm

Co-authored-by: Mark Suckerberg <mark@suckerberg.gay>
Signed-off-by: Bjarl <94164348+Bjarl@users.noreply.github.com>

Erikafox committed:
Update anomalies_hallucination.dm

Erikafox committed:
changes

Erikafox committed:
Merge branch 'planetary-anomalies' of https://github.com/Bjarl/Shiptest into planetary-anomalies

MarkSuckerberg reviewed (approved):
looks legit

MarkSuckerberg commented:
oop

Erikafox commented:
AUGH

Erikafox committed:
Merge remote-tracking branch 'upstream/master' into planetary-anomalies

Erikafox committed:
Update anomalies_hallucination.dm

Zevotech commented on code/game/objects/effects/anomalies/_anomalies.dm:
```diff
@@ -0,0 +1,148 @@
+/obj/effect/anomaly/attackby(obj/item/weapon, mob/user, params)
+	if(weapon.tool_behaviour == TOOL_ANALYZER && aSignal)
+		to_chat(user, span_notice("Analyzing..."))
+		if(do_after(user, 20, TRUE, src))
```
This makes some anomalies near impossible to do (Ex. Throngler) as them moving you or proning you will cancel the do_after and make you unable to analyze it without cheesing it with some sort of obstacle. The throngler pulses before you can finish analyzing it, which cancels the do_after. If you're not going to make it instant, make the "Analyzing..." text something more like "You start analyzing the [src]'s fields..."

Erikafox replied:
I'm going to add a quicker way to scan in #2103, anomaly neutralizers also remain an option, but I'll go ahead and change the text.

Erikafox committed:
Update _anomalies.dm

Mirag1993 commented on code/game/objects/effects/anomalies/anomalies_gravity.dm:
```diff
@@ -0,0 +1,122 @@
+
+/obj/effect/anomaly/grav/proc/gravShock(mob/living/Guy)
+	if(boing && isliving(Guy) && !Guy.stat)
+		Guy.Paralyze(40)
```
why without  COOLDOWN ????

Erikafox replied:
I dunno

Mirag1993 replied:
> I dunno

It's firing multiple times with no cooldown (like 0.001s intervals). Should probably add a cooldown check to prevent spam effects that can kill players.

Erikafox replied:
Hasn't been an issue for us bro. This pr is over 2 years old.

Mirag1993 replied:
Saying “it’s old” or “hasn’t been an issue” is just nonsense — that’s not how bugs work. I’ve pointed it out, do whatever you want with that info.

rivercadaver replied:
i’ve never seen a throngler proc more than once every 5 or so seconds unless this is a diff anomaly

FalloutFalcon replied:
It fires as apart of SSobj so it acctually fires every 2 seconds. nothing in the game fires at that high of a interval, the lowest you can go is functionally 0.1s

Mirag1993 replied:
> It fires as apart of SSobj so it acctually fires every 2 seconds. nothing in the game fires at that high of a interval, the lowest you can go is functionally 0.1s

Correct me if I'm wrong. Signals fire immediately when objects enter the anomaly's area? Without cooldown?
